File: backend/Models/Product.cs
namespace backend.Models;

public record DbProduct
{
    public int Id { get; init; }
    public DateTime CreatedDateTime { get; set; }
    public int CreatedByUserId { get; set; }
    public int Order { get; init; }
    public string Name { get; init; } = "";
    public decimal Price { get; init; }
    public bool InStock { get; set; }
}

public record NewProduct
{
    public required string Name { get; init; }
    public decimal Price { get; init; }
}

public record UpdateProduct
{
    public required int Id { get; init; }
    public required string Name { get; init; }
    public required decimal Price { get; init; }
}

public record UpdateProductInStock
{
    public required int Id { get; init; }
    public required bool InStock { get; init; }
}

public record UpdateProductOrder
{
    public required int Id { get; init; }
    public required int Order { get; init; }
}

File: frontend/src/backendDto/product.ts
export interface ProductDto {
    id: number;
    createdDateTime: string;
    createdByUserId: number;
    order: number;
    name: string;
    price: number;
    inStock: boolean;
}

export interface NewProductDto {
    name: string;
    price: number;
}

export interface UpdateProductDto {
    id: number;
    name: string;
    price: number;
}

export interface UpdateProductInStockDto {
    id: number;
    inStock: boolean;
}

export interface UpdateProductOrderDto {
    id: number;
    order: number;
}
